packages/realm-server/tests/realm-server-test.ts

```diff
@@ -2135,6 +2135,104 @@ module('Realm Server', function (hooks) {
       });
     });
   });
+
+  module('card css source GET request', function (_hooks) {
+    module('public readable realm', function (hooks) {
+      setupPermissionedRealm(hooks, {
+        '*': ['read'],
+      });
+
+      test('serves the request css for card that has css indexed', async function (assert) {
+        let response = await request
+          .get('/fancy-person')
+          .set('Accept', 'application/vnd.card+css');
+
+        assert.strictEqual(response.status, 200, 'HTTP 200 status');
+        assert.notEqual(response.headers.etag, null, 'etag exists');
+        assert.strictEqual(
+          response.headers['cache-control'],
+          'must-revalidate',
+          'cache-control is correct',
+        );
+        assert.strictEqual(
+          response.text.replace(/\s+/g, ' '),
+          `.fancy-border[data-scopedcss-b9c3801d7f-403fcb7205] { border: 1px solid pink; }`,
+          'CSS is correct',
+        );
+      });
+
+      test('serves the request css for card that has no css indexed', async function (assert) {
+        let response = await request
+          .get('/person')
+          .set('Accept', 'application/vnd.card+css');
+
+        assert.strictEqual(response.status, 404, 'HTTP 404 status');
+      });
+
+      test('respects client etag', async function (assert) {
+        let response = await request
+          .head('/fancy-person')
+          .set('Accept', 'application/vnd.card+css');
+
+        let etag = response.headers.etag;
+        response = await request
+          .get('/fancy-person')
+          .set('Accept', 'application/vnd.card+css')
+          .set('If-None-Match', etag);
+
+        // 304 Not Modified
+        assert.strictEqual(response.status, 304, 'HTTP 304 status');
+        assert.deepEqual(response.body, {}, 'body is empty');
+      });
+    });
+
+    module('permissioned realm', function (hooks) {
```

Comment:
There was some discussion whether this endpoint should be behind auth or not, and my reasoning to enforce auth to read CSS from non-public realms is that since CSS is a part of a module, and reading a module needs permission, then reading CSS should behave the same...